modules/general_functions.py

```diff
@@ -656,3 +656,25 @@ def find_next(iterable: Iterable[T], cond: Callable[[T], bool]) -> T:
         return next(i for i in iterable if cond(i))
     except StopIteration:
         raise ValueError("Value not found in iterable.")
+
+
+def check_if_sum_spectrum_is_selected(directory):
+    """
+    Check if the sum spectrum is selected on the GUI.
+
+    Args:
+        directory: Directory that is iterated
+
+    Return:
+        True if there is at least one file (path) that contains "SUM" string.
+        False if there is not.
+
+    """
+    measured_sum = False
+    simulated_sum = False
+    for filepath in os.listdir(directory):
+        if "MEASURED_SUM" in filepath:
+            measured_sum = True
+        if "SIMULATED_SUM" in filepath:
+            simulated_sum = True
+    return measured_sum, simulated_sum
```

Comment:
```suggestion
    measured_sum_found = False
    simulated_sum_found = False
    with os.scandir(directory) as sdir:
        for entry in sdir:
            if entry.name.startswith("MEASURED_SUM"):
                measured_sum_found = True
            if entry.name.startswith("SIMULATED_SUM"):
                simulated_sum_found = True
    return measured_sum_found, simulated_sum_found

```

Improvements:
- checking for "MEASURED_SUM"/"SIMULATED_SUM" anywhere in the path could lead to false positives, entry.name is less likely to do so
- `os.scandir` is faster and always returns a path-like object, unlike `os.listdir` which is more complicated
- more descriptive variable names

Also, those "MEASURED_SUM" and "SIMULATED_SUM" strings should be constants that are defined in one place (not inside the function).

Reply:
OK